cache/cache.go

```diff
@@ -37,17 +43,21 @@ func BasePath(checksum, uri string) string {
 //
 // "fastFailClient" is a HTTP client configured to fail quickly if a remote
 // server is unavailable, for use in optional checks.
-func Open(root string, ghClient *github.Client, client *http.Client, fastFailClient *http.Client) (*Cache, error) {
+//
+// "strategies" are used to download URLS, attempted in order.
+func Open(root string, strategies []DownloadStrategy, client *http.Client, fastFailClient *http.Client) (*Cache, error) {
 	err := os.MkdirAll(root, os.ModePerm)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
-	return &Cache{
+	c := &Cache{
 		root:               root,
 		httpClient:         client,
-		gh:                 ghClient,
 		fastFailHTTPClient: fastFailClient,
-	}, nil
+	}
+	c.strategies = append(c.strategies, c.defaultDownloadStrategy)
```

Comment:
Feels strange that we have to add `c.defaultDownloadStrategy` to `c.strategies` at initialisation. Maybe this logic should be either completely internal to `Cache`, or `defaultDownloadStrategy` should not be in `Cache`?

Reply:
What do you mean, it is completely internal?

Reply:
Ah, sorry, for some reason I didn't notice this was in the cache package. Disregard!